src/java.net.http/share/classes/jdk/internal/net/http/RequestPublishers.java

```diff
@@ -229,74 +223,37 @@ public void subscribe(Flow.Subscriber<? super ByteBuffer> subscriber) {
 
     /**
      * Publishes the content of a given file.
-     * <p>
-     * Privileged actions are performed within a limited doPrivileged that only
-     * asserts the specific, read, file permission that was checked during the
-     * construction of this FilePublisher. This only applies if the file system
-     * that created the file provides interoperability with {@code java.io.File}.
      */
     public static class FilePublisher implements BodyPublisher {
 
         private final Path path;
         private final long length;
         private final Function<Path, InputStream> inputStreamSupplier;
 
-        private static String pathForSecurityCheck(Path path) {
+        private static String pathForDefaultFSCheck(Path path) {
             return path.toFile().getPath();
         }
 
         /**
          * Factory for creating FilePublisher.
-         *
-         * Permission checks are performed here before construction of the
-         * FilePublisher. Permission checking and construction are deliberately
-         * and tightly co-located.
          */
         public static FilePublisher create(Path path)
                 throws FileNotFoundException {
-            @SuppressWarnings("removal")
-            SecurityManager sm = System.getSecurityManager();
-            FilePermission filePermission = null;
             boolean defaultFS = true;
 
             try {
-                String fn = pathForSecurityCheck(path);
-                if (sm != null) {
-                    FilePermission readPermission = new FilePermission(fn, "read");
-                    sm.checkPermission(readPermission);
-                    filePermission = readPermission;
-                }
+                pathForDefaultFSCheck(path);
```

Comment:
Is this intended, calling the method but ignoring the return value? If so, a comment might be in order.

Reply:
The method might throw UOE if the path doesn't come from the default file system. I didn't want to investigate and change that logic now - I wasn't sure I fully understood why we have special cases here, so decided to keep it.